public/components/trace_analytics/components/traces/span_detail_panel.tsx

```diff
@@ -55,6 +60,50 @@ export function SpanDetailPanel(props: {
   } else {
     [data, setData] = [localData, localSetData];
   }
+  const fullRange = [0, data.ganttMaxX * 1.1];
+  const [selectedRange, setSelectedRange] = useState(fullRange);
+  const isLocked = useObservable(chrome!.getIsNavDrawerLocked$() ?? false);
+  const [isFullScreen, setIsFullScreen] = useState(false);
+  const containerRef = useRef<HTMLDivElement | null>(null);
+  const [availableWidth, setAvailableWidth] = useState<number>(window.innerWidth);
+
+  useEffect(() => {
+    const updateAvailableWidth = () => {
+      if (containerRef.current) {
+        setAvailableWidth(containerRef.current.getBoundingClientRect().width);
+      } else {
+        setAvailableWidth(window.innerWidth);
+      }
+    };
+
+    const handleFullScreenChange = () => {
+      const isFullscreenActive = !!document.fullscreenElement;
+      setIsFullScreen(isFullscreenActive);
+      updateAvailableWidth();
+    };
```

Comment:
Nit. I personally don't ride with adding the function definitions inside useEffect(). But is not a blocker.